internal/engine/engine_test.go

```diff
@@ -277,3 +356,40 @@ var _ = Describe("Check Name Queries", func() {
 		})
 	})
 })
+
+// writePaddedTarball writes a tar archive to out with filename containing contents at the base path
+// with extra bytes written at the end of length extraBytes+1.
+// note: this should only be used as a helper function in tests
+func writePaddedTarball(out io.Writer, contents []byte, filename string, extraBytes uint) error {
+	tw := tar.NewWriter(out)
+	defer tw.Close()
+
+	header := &tar.Header{
+		Typeflag: tar.TypeReg,
+		Name:     filename,
+		Size:     int64(len(contents)),
+		Mode:     420,
+		Format:   tar.FormatPAX,
+	}
+
+	err := tw.WriteHeader(header)
+	if err != nil {
+		return err
+	}
+
+	_, err = io.Copy(tw, bytes.NewReader(contents))
+	if err != nil {
+		return err
+	}
+
+	if extraBytes > 0 {
```

Comment:
This conditional seems out of place. If extraBytes `<=` 0, then we don't write any padding which defeats the purpose of this function, no?

Reply:
I see that you're using this function to write a tarball that isn't padded as well. In that case, maybe just remove the addition of 1 extra byte, rename the function, update the comment, etc.